base/abstractarray.jl

```diff
@@ -246,8 +246,8 @@ end
 ## Traits for array types ##
 
 abstract LinearIndexing
-immutable LinearFast <: LinearIndexing end
-immutable LinearSlow <: LinearIndexing end
+struct LinearFast <: LinearIndexing end
+struct LinearSlow <: LinearIndexing end
```

Comment:
Zero is a valid number of fields :) Most (all?) languages with structs allow empty structs, and they're especially useful in languages that support various type system tricks.